### DISCUSSION TITLE
option to Group Ingredients for recipes that have multiple parts to the recipe

### DISCUSSION THREAD

**Discussion by MactasticMendez** [mealie-recipes/mealie#4778] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

this would address long ingredient lists for recipes with multiple parts and also resolve the issue where if multiple parts use the same ingredient it can be not as clear as it should be to users which quantity of that ingredient is needed for that part of the recipe

### Please provide a concise description of the feature that would resolve your issue.

To resolve this issue a user should be able to tag an ingredient in the ingredients list with what part of the recipe it is for. For instance if a cake recipe has sugar in both the cake and icing you should be able to add one item of sugar tagged as cake and another item of sugar tagged as butter cream for instance.

These ingredients would then be grouped together so that all the cake ingredients are together and all the butter cream ingredients are together as well.

- CAKE Batter
ingredient 1
ingredient 2
ingredient 3

- butter cream
ingredient 1
ingredient 2

### Please consider and list out some caveats or tradeoffs made in your design decision

trade offs could be that its another field for users to populate but it could be made optional.

also there is the argument that you could split up each part of the recipe into its own but that could make it harder for novice bakers or cooks who might not know what other recipes are needed as part of the overall cook or quantities are need for those in a scaled recipe. So by having it together but grouped in the same recipe could make it simpler and easier to view and cook with.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by hay-kot** [comment]
Hey! This is already a feature, check out the "..." menus on the Ingredients and Instructions

![CleanShot 2024-12-25 at 17 12 26@2x](https://github.com/user-attachments/assets/5288ea22-d2a7-4d30-92d9-12752a2e028d)

![CleanShot 2024-12-25 at 17 13 12@2x](https://github.com/user-attachments/assets/43ada84e-67fc-42a7-a50f-a4fc02949596)
